app/src/main/java/org/stepik/android/presentation/course_list/CourseListCollectionPresenter.kt

```diff
@@ -25,7 +27,9 @@ constructor(
 
     viewContainer: PresenterViewContainer<CourseListView>,
     continueCoursePresenterDelegate: CourseContinuePresenterDelegateImpl
-) : PresenterBase<CourseListView>(viewContainer), CourseContinuePresenterDelegate by continueCoursePresenterDelegate {
+) : PresenterBase<CourseListView>(viewContainer),
```

Comment:
why not to create state for `CourseListCollectionView` with course collection data with nested `CourseListView.State`?